hcloud/action_test.go

```diff
@@ -134,3 +134,76 @@ func TestActionClientAll(t *testing.T) {
 		t.Errorf("unexpected actions")
 	}
 }
+
+func TestActionClientWait(t *testing.T) {
+	env := newTestEnv()
+	defer env.Teardown()
+
+	callCount := 0
+	env.Mux.HandleFunc("/actions/1", func(w http.ResponseWriter, r *http.Request) {
+		callCount++
+		w.Header().Set("Content-Type", "application/json")
+		switch callCount {
+		case 1:
+			json.NewEncoder(w).Encode(schema.ActionGetResponse{
+				Action: schema.Action{
+					ID:       1,
+					Status:   "running",
+					Progress: 50,
+				},
+			})
+		case 2:
+			w.WriteHeader(http.StatusTooManyRequests)
+			json.NewEncoder(w).Encode(schema.ErrorResponse{
+				Error: schema.Error{
+					Code:    "limit_reached",
+					Message: "ratelimited",
+				},
+			})
+			return
+		case 3:
+			json.NewEncoder(w).Encode(schema.ActionGetResponse{
+				Action: schema.Action{
+					ID:       1,
+					Status:   "error",
+					Progress: 100,
+					Error: &schema.ActionError{
+						Code:    "action_failed",
+						Message: "action failed",
+					},
+				},
+			})
+		default:
+			t.Errorf("unexpected number of calls to the test server: %v", callCount)
+		}
+	})
+	action := &Action{
+		ID:       1,
+		Status:   ActionStatusRunning,
+		Progress: 0,
+	}
+
+	ctx := context.Background()
+	progressCh, errCh := env.Client.Action.Wait(ctx, action)
+	var (
+		progressUpdates []int
+		err             error
+	)
+
+loop:
+	for {
+		select {
+		case progress := <-progressCh:
+			progressUpdates = append(progressUpdates, progress)
+		case err = <-errCh:
+			break loop
+		}
+	}
+
+	if err == nil {
+		t.Fatalf("error expected %v", err)
```

Comment:
Wrong message?

Reply:
Is correct because i expect an non nil `err` because the state of the action is error

Reply:
Your condition might be correct, but why would you print `error expected <nil>`?